pkgs/by-name/va/valkey/package.nix

```diff
@@ -0,0 +1,84 @@
+{ lib, stdenv, fetchFromGitHub, fetchurl, fetchpatch, lua, jemalloc, pkg-config, nixosTests
+, tcl, which, ps, getconf
+, withSystemd ? lib.meta.availableOn stdenv.hostPlatform systemd, systemd
+# dependency ordering is broken at the moment when building with openssl
+, tlsSupport ? !stdenv.hostPlatform.isStatic, openssl
+
+# Using system jemalloc fixes cross-compilation and various setups.
+# However the experimental 'active defragmentation' feature of valkey requires
+# their custom patched version of jemalloc.
+, useSystemJemalloc ? true
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "valkey";
+  version = "7.2.4-rc1";
+
+  src = fetchFromGitHub {
+    owner = "valkey-io";
+    repo = "valkey";
+    rev = finalAttrs.version;
```

Comment:
I don't think there's any real reason to do this as if this gets overrided, the hash will change.

Reply:
You are probably right. There is current discussion on how to handle the versioning... at some point something will be decided... but for now I think it doesn't matter much.